### ISSUE TITLE
spirv-val does not validate invalid ExtractComposite instruction

### ISSUE THREAD

**Issue by danginsburg** [KhronosGroup/SPIRV-Tools#447]
The attached SPIR-V contains invalid SPIR-V:

```
   %3475 = OpCompositeExtract %6 %3473 0
   %3476 = OpCompositeExtract %6 %3475 0
```

The second extract is trying to extract a float from a float (not legal)
%6 is type float.

A validator for SPIR-V should validate section 2.2.2 of the spec where it defines
what a composite is, and, couple that with the “ExtractComposite” instruction that
only operates on composite types.

**Comment by danginsburg** [comment]
SPIR-V file containing the error is located here: https://app.box.com/s/do85guirrzjir017igi7p4vwl2n2n63t

**Comment by umar456** [comment]
Hi Dan,

The validator is a work in progress and the type checking is not implemented in the current version.

**Comment by ehsannas** [comment]
I have actually fixed this via commit d1868b93618181cd765ea16aa4aec8de9f4d3b2b

I just ran the attached .spv file and got the following error (as expected):
`error: 23182: OpCompositeExtract reached non-composite type while indexes still remain to be traversed.`
